File: android/src/main/kotlin/com/emarsys/emarsys_sdk/commands/ClearContactCommand.kt
package com.emarsys.emarsys_sdk.commands

import com.emarsys.Emarsys
import com.emarsys.core.api.result.CompletionListener
import com.emarsys.emarsys_sdk.EmarsysCommand


class ClearContactCommand : EmarsysCommand {

    override fun execute(parameters: Map<String, Any?>?, resultCallback: ResultCallback) {
        Emarsys.clearContact(CompletionListener {
            resultCallback.invoke(null, it)
        })
    }

    override fun equals(other: Any?): Boolean {
        if (this === other) return true
        if (javaClass != other?.javaClass) return false
        return true
    }

    override fun hashCode(): Int {
        return javaClass.hashCode()
    }
}

File: ios/Classes/Wrapper/Commands/Contact/ClearContactCommand.swift
//
// Created by Emarsys on 2021. 04. 22..
//
import EmarsysSDK

public class ClearContactCommand: EmarsysCommandProtocol {

    func execute(arguments: [String : Any]?, resultCallback: @escaping ResultCallback) {
        Emarsys.clearContact() { error in
            if let e = error {
                resultCallback(["error": e])
            } else {
                resultCallback(["success": true])
            }
        }
    }

}
